### PR TITLE
[libtins] Misc fixes

### PR DESCRIPTION
**Pull request by dg0yt**
- #### What does your PR fix?  
  Fixes source writes which break parallel configure. (Needed for #21507.)
  Adds explicit openssl dependency.
  Switches from winpcap to libpcap, for all platforms. (Note that winpcap is skipped in CI.)
  Disables documentation.
  Marks uwp unsuppported. (No pcap implementation seems to support this platform.)

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  all but uwp, no

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  yes

### CODE REVIEW

**Comment by dg0yt** [comment]
Still cascading on osx. Really no osx support in libpcap?

File: ports/libtins/vcpkg.json [hunk 1]
```diff
@@ -1,22 +1,21 @@
 {
   "name": "libtins",
   "version-string": "4.3",
```
Comment (JonLiu1993) [inline ports/libtins/vcpkg.json:3]:
```suggestion
  "version": "4.3",
```

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Comment by JonLiu1993** [comment]
@dg0yt ,Thanks for your pr, this is error log:
https://dev.azure.com/vcpkg/public/_build/results?buildId=63987&view=artifacts&pathAsName=false&type=publishedArtifacts

**Review by JackBoosY — APPROVED** [review]

**Review by JonLiu1993 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks!

**Pull request merged**